libs/execution_context/ebpf_program.c

```diff
@@ -459,15 +459,93 @@ ebpf_program_get_helper_function_address(const ebpf_program_t* program, uint32_t
 }
 
 ebpf_result_t
-ebpf_program_get_program_info_data(const ebpf_program_t* program, const ebpf_extension_data_t** program_info_data)
+ebpf_program_get_program_info_data(_In_ const ebpf_program_t* program, _Outptr_ ebpf_program_info_t** program_info)
 {
-    if (program->program_invalidated)
-        return EBPF_EXTENSION_FAILED_TO_LOAD;
+    ebpf_result_t result = EBPF_SUCCESS;
+    ebpf_program_data_t* program_data = NULL;
+    ebpf_program_data_t* global_helper_program_data = NULL;
+    ebpf_program_info_t* local_program_info = NULL;
+    uint32_t total_count_of_helpers = 0;
+    uint32_t helper_index = 0;
 
-    if (!program->program_info_data)
-        return EBPF_EXTENSION_FAILED_TO_LOAD;
+    if (program_info == NULL) {
+        result = EBPF_INVALID_ARGUMENT;
+        goto Exit;
+    }
+    *program_info = NULL;
 
-    *program_info_data = program->program_info_data;
+    if (program->program_invalidated) {
+        result = EBPF_EXTENSION_FAILED_TO_LOAD;
+        goto Exit;
+    }
 
-    return EBPF_SUCCESS;
+    if (!program->program_info_data) {
```

Comment:
where is the `program_info_data` field defined?  I can't find it.  It seems like it needs to be renamed.

Reply:
line 43 of ebpf_program.c
`const ebpf_extension_data_t* program_info_data;`
What would you like this field to be renamed to? extension_data seems too generic.

Reply:
The field name suffix should match the type suffix.  So anything that ends in "extension_data".

Reply:
renamed it to program_info_provider_data as in program_info NPI provider's data - which is truly what it is.